last_practice5_2.java

```diff
@@ -0,0 +1,63 @@
+import java.util.Scanner;
+
+public class last_practice5_2 {
+
+	public static boolean isPrime(int number) {
+
+		// All negative numbers, 0 and 1
+		// are not a prime number
+		if (number <= 1)
+			return false;
+
+		// check for remaining
+		for (int i = 2; i < number; i++)
+			if (number % i == 0)
+				return false;
+
+		return true;
+	}
+
+	public static boolean isPerfect(int number) {
+
+		int temp = 0;
+		for (int i = 1; i <= number / 2; i++) {
+			if (number % i == 0) {
+				temp += i;
+			}
+			if (temp == number) {
+
+				return true;
+
+			}
+		}
+		return false;
+
+	}
+
+	public static void main(String[] args) {
+
+		// declare variables
+		int number = 0;
+		boolean flag = false;
+
+		// create Scanner class object
+		Scanner scan = new Scanner(System.in);
+
+		// read number
+		System.out.print("Enter a number: ");
+		number = scan.nextInt();
+
+		// check number is prime number or not
+		flag = isPrime(number);

```

Comment:
that's not what is being asked.
You need to ask the user wether he wants to know if it's a prime or a perfect number (store that in a variable userChoice). Then ask for the number (variable number).
Then if userChoice==prime => call isprime, if userChoice==perfect => call isPerfect, else quit

Reply:
I totally missed understand it. 
I'll basically re-do it.